src/components/ResultCard/ResultCard.tsx

```diff
@@ -1,31 +1,71 @@
-import React, { KeyboardEvent } from 'react';
+import React, { KeyboardEvent, useContext } from 'react';
 import { BrowseResultData } from '../../types';
 import ResultCtaButton from '../ResultCtaButton/ResultCtaButton';
+import QuizContext from '../CioQuiz/context';
 
 interface ResultCardProps {
   result: Partial<BrowseResultData>;
   addToCartCallback: (clickedResult: Partial<BrowseResultData>) => any;
   clickItemCallback?: (clickedResult: Partial<BrowseResultData>) => any;
   salePriceKey?: string;
   regularPriceKey?: string;
+  resultPosition: number;
 }
 
 export default function ResultCard(props: ResultCardProps) {
-  const { result, addToCartCallback, clickItemCallback, salePriceKey, regularPriceKey } = props;
+  const {
+    result,
+    addToCartCallback,
+    clickItemCallback: customClickItemCallback,
+    salePriceKey,
+    regularPriceKey,
+    resultPosition,
+  } = props;
+  const { resultsResponse } = useContext(QuizContext);
+  const { cioClient } = useContext(QuizContext);
   const salePrice = salePriceKey && result?.data?.[salePriceKey];
   const regularPrice = regularPriceKey && result?.data?.[regularPriceKey];
 
-  const clickHandler = () => {
-    if (clickItemCallback && typeof clickItemCallback === 'function') {
-      clickItemCallback(result);
+  const clickItemCallback = () => {
+    /* eslint-disable @typescript-eslint/naming-convention */
+    if (resultsResponse && resultsResponse.request && resultsResponse.response) {
```

Comment:
This code partially repeats. I thought about moving the shared parts to a separate function but I think there is enough difference that makes it enough to keep this as is. Let me know what you guys think